core/src/avm1/globals/automation_action_generator.rs

```diff
@@ -0,0 +1,73 @@
+use crate::avm1::activation::Activation;
+use crate::avm1::error::Error;
+use crate::avm1::property_decl::{DeclContext, Declaration, SystemClass};
+use crate::avm1::{Object, Value};
+use crate::avm1_stub;
+
+const PROTO_DECLS: &[Declaration] = declare_properties! {
+    "generateAction" => method(generate_action);
+    "generateActions" => method(generate_actions);
+    "valueOf" => method(value_of);
+    "toString" => method(to_string);
+};
+
+pub fn create_class<'gc>(
+    context: &mut DeclContext<'_, 'gc>,
+    super_proto: Object<'gc>,
+) -> SystemClass<'gc> {
+    let class = context.class(constructor, super_proto);
```

Comment:
It'd be cleaner to use `context.empty_class` instead of an explicit empty constructor.

Reply:
Done